components/script/dom/document.rs

```diff
@@ -515,18 +515,38 @@ impl Document {
     /// Attempt to find a named element in this page's document.
     /// https://html.spec.whatwg.org/multipage/#the-indicated-part-of-the-document
     pub fn find_fragment_node(&self, fragid: &str) -> Option<Root<Element>> {
-        self.get_element_by_id(&Atom::from(fragid)).or_else(|| {
-            let check_anchor = |node: &HTMLAnchorElement| {
-                let elem = node.upcast::<Element>();
-                elem.get_attribute(&ns!(), &atom!("name"))
-                    .map_or(false, |attr| &**attr.value() == fragid)
-            };
-            let doc_node = self.upcast::<Node>();
-            doc_node.traverse_preorder()
-                    .filter_map(Root::downcast)
-                    .find(|node| check_anchor(&node))
-                    .map(Root::upcast)
-        })
+        // Step 2
```

Comment:
`// Step 1 is not handled here; the fragid is already obtained by the calling function`

That ok? I think including the name of the file/function that obtains the fragid isn't a good idea, for maintainability reasons. It would be in danger of going out of date if it the other code changed without affecting this location.